addons/website_sale_picking/models/stock_quant.py

```diff
@@ -0,0 +1,17 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.exceptions import ValidationError
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    def check_location_id(self):
+        # Since onsite picking is located on warehouses, disable the check on these for onsite picking.
+        onsite_product = self.env.ref('website_sale_picking.onsite_delivery_product')
+        for record in self:
+            if record.location_id.usage == 'view' and record.product_id.id != onsite_product.id:
+                raise ValidationError(
+                    _('You cannot take products from or deliver products to a location of type "view" (%s).') % record.location_id.name)
```

Comment:
I am guessing this allows deliveries to warehouses?
This is a dangerous override, it might break custom modules.
I would instead change the original one to add a whitelist of products and have something such as:
```suggestion
    def _get_view_product_whitelist(self):
        onsite_delivery_product = self.env.ref('website_sale_picking.onsite_delivery_product', raise_if_not_found=False) or self.env['product.product']
        return super()._get_view_product_whitelist() + onsite_delivery_product
```

Also note that `record.product_id.id != onsite_product.id` is equivalent to `record.product_id != onsite_product`